aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -231,18 +277,61 @@ Tensor linalg_pinv(const Tensor& input, const Tensor& rcond, bool hermitian) {
     Tensor S_abs = S.abs();
     // eigenvalues are sorted in ascending order starting with negative values, we need a maximum value of abs(eigenvalues)
     Tensor max_val = S_abs.amax(/*dim=*/-1, /*keepdim=*/true);
-    Tensor S_pseudoinv = at::where(S_abs > (rcond.unsqueeze(-1) * max_val), S.reciprocal(), at::zeros({}, S.options())).to(input.dtype());
+    Tensor tol = at::max(atol.unsqueeze(-1), rtol.unsqueeze(-1) * max_val);
+    Tensor S_pseudoinv = at::where(S_abs > tol, S.reciprocal(), at::zeros({}, S.options())).to(input.dtype());
     // computes U @ diag(S_pseudoinv) @ U.conj().T
     return at::matmul(U * S_pseudoinv.unsqueeze(-2), U.conj().transpose(-2, -1));
   }
 }
 
+Tensor linalg_pinv(const Tensor& input, double atol, double rtol, bool hermitian) {
+  Tensor atol_tensor, rtol_tensor;
+  std::tie(atol_tensor, rtol_tensor) = get_atol_rtol(input, atol, rtol);
+  return at::linalg_pinv(input, atol_tensor, rtol_tensor, hermitian);
+}
+
+Tensor linalg_pinv(const Tensor& input, const Tensor& rcond, bool hermitian) {
+  // For NumPy compatibility the rcond argument is used as relative tolerance
+  TORCH_CHECK(!at::isComplexType(rcond.scalar_type()),
+              "torch.linalg.pinv: rcond tensor of complex type is not supported.");
```

Comment:
Could you factor out all these errors into `LinearAlgebraUtils.h` for consistency and easy of use? Same for `atol` and `acond` and so on.

Reply:
Something like `checkNotComplex`? But this error check occurs in the code only in this function.

Reply:
It's for these two functions and also `lstsq` whenever we support it, right?

Reply:
Done.